package modelo;

import java.util.Objects;

public class Libro {

    private int id;

    private String titulo;

    private String autor;

    private int anio;

    private String genero;

    // Constructor principal.
    public Libro(int id, String titulo, String autor, int anio, String genero) {
        this.id = id;
        this.titulo = titulo;
        this.autor = autor;
        this.anio = anio;
        this.genero = genero;
    }

    // Constructor usado cuando quiero insertar un objeto en la base de datos que omite el id ya que es autoincremental
    // en la base de datos.
    public Libro(String titulo, String autor, int anio, String genero) {
        this.titulo = titulo;
        this.autor = autor;
        this.anio = anio;
        this.genero = genero;
    }

    public int getId() {
        return id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public int getAnio() {
        return anio;
    }

    public void setAnio(int anio) {
        this.anio = anio;
    }

    public String getGenero() {
        return genero;
    }

    public void setGenero(String genero) {
        this.genero = genero;
    }

    @Override
    public String toString() {
        return this.id + " - " + this.titulo +
                "\n Autor : " + autor +
                "\n Anio publicacion : " + anio +
                "\n Genero : " + genero + "\n";
    }

    // Necesito sobreescribir el metodo equals para que obvie el atributo id entre un objeto y otro cuando se compare.
    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Libro libro = (Libro) o;
        return anio == libro.anio && Objects.equals(titulo, libro.titulo) && Objects.equals(autor, libro.autor) && Objects.equals(genero, libro.genero);
    }

    @Override
    public int hashCode() {
        return Objects.hash(titulo, autor, anio, genero);
    }
}
